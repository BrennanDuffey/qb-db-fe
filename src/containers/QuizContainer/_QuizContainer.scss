.quiz-card {
  min-height: 450px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  .quiz-text {
    width: 80%;
    line-height: 1.5;
    margin: 20px auto;
    text-align: left;
    font-size: 1.1rem;
  }
  .quiz-button {
    background-color: rgba(0, 0, 0, .1);
    padding: 10px;
    border: solid 1px rgba(0, 0, 0, .15);
    border-radius: 12px;
    width: 60%;
    margin: 0 auto;
    &:hover {
      border: solid 1px rgba(0, 0, 0, .25);
      background-color: rgba(0, 0, 0, .2);
    }
  }
}