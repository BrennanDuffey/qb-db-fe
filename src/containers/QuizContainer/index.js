import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Redirect } from "react-router-dom";
import PropTypes from "prop-types";


export class QuizContainer extends Component {
  constructor() {
    super();
    this.state = {
      questionCounter: 0,
      displayAnswer: false
    }
  }

  incrementCounter = async () => {
    const questionCounter = this.state.questionCounter + 1;
    await this.setState({ questionCounter });
    this.toggleDisplay();
  }

  toggleDisplay = () => {
    this.setState({ displayAnswer: !this.state.displayAnswer })
  }

  createMarkup(text) {
    return {__html: text}
  }


  render() {
    const { tossups } = this.props
    const { displayAnswer, questionCounter } = this.state
    if (!tossups.length || tossups.length === questionCounter) {
      return <Redirect to={'/Controls'}/>
    }
    return (
      <section className="quiz-card">
        {!displayAnswer && <div className="quiz-text" dangerouslySetInnerHTML={this.createMarkup(tossups[questionCounter].formatted_text)} />}
        {!displayAnswer && <h3 className="quiz-button" onClick={this.toggleDisplay}>Show Answer </h3>}
        {displayAnswer && <div className="quiz-text" dangerouslySetInnerHTML={this.createMarkup(tossups[questionCounter].formatted_answer)} />}
        {displayAnswer && <h3 className="quiz-button" onClick={() => {this.incrementCounter()}}>Next Question</h3>}
      </section>
    )
  }
}

export const mapStateToProps = (state) => ({
  tossups: state.tossups
});

QuizContainer.propTypes = {
  dispatch: PropTypes.func,
  history: PropTypes.object,
  location: PropTypes.object,
  match: PropTypes.object,
  tossups: PropTypes.array
}

export default connect(mapStateToProps)(QuizContainer)