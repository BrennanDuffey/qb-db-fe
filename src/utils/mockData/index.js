export const mockTossups = [
  {
    "id": 123902,
    "text": "Description acceptable. A paper by Urribarri, Rios, and Ferrer hoped that treatment with ammonia would allow this process to be used as a source of raw vegetable protein for the feeding of livestock. Pollution with poliaspartic acid and disruption of the hypolemnitic cone have been suggested as causes of this phenomenon, which may result in lower (*) oxygen levels that threaten the curbina fishery. Researchers at the University of Zulia blamed ships from Florida or Texas for starting this phenomenon by transporting an  ► invasive species of genus Lemna. For 10 points, what bloom of aquatic plants threatens the ecological health of a certain lake in western Venezuela?",
    "answer": "duckweed infestation of Lake Maracaibo (accept equivalents)",
    "number": 3,
    "tournament_id": null,
    "category_id": 20,
    "subcategory_id": null,
    "round": "[missing]",
    "created_at": "2017-08-27T22:44:27.515Z",
    "updated_at": "2017-09-01T07:10:28.156Z",
    "quinterest_id": 64463,
    "formatted_text": "<b>Description acceptable. A paper by Urribarri, Rios, and Ferrer hoped that treatment with ammonia would allow this process to be used as a source of raw vegetable protein for the feeding of livestock. Pollution with poliaspartic acid and disruption of the hypolemnitic cone have been suggested as causes of this phenomenon, which may result in lower (*)</b>) oxygen levels that threaten the curbina fishery. Researchers at the University of Zulia blamed ships from Florida or Texas for starting this phenomenon by transporting an  &#9658; invasive species of genus Lemna. For 10 points, what bloom of aquatic plants threatens the ecological health of a certain lake in western Venezuela?",
    "formatted_answer": "duckweed infestation of Lake Maracaibo (accept equivalents)",
    "wikipedia_url": null,
    "url": "https://www.quizdb.org/api/tossups/123902",
    "type": "tossup",
    "tournament": {
    },
    "category": {
      "id": 20,
      "name": "Geography",
      "created_at": "2017-08-27T19:30:45.889Z",
      "updated_at": "2017-08-27T19:30:45.889Z",
      "url": "https://www.quizdb.org/api/categories/20"
    },
    "subcategory": {
    }
  },
  {
    "id": 123907,
    "text": "The \"Spreuer\" example of this feature in Switzerland contains Kaspar Meglinger's 17th-century paintings of the Danse Macabre. Examples of these features from Qing Dynasty China include the \"stepping toad\" and the \"eternal celebration.\" The longest feature of this type in the world, at Hartland in New Brunswick, was damaged by ice in the 1920s. The \"Burr (*) arch\" is a typical design for these structures, which exist at Kaufman's Distillery, White Rock Forge, and Landis Mill in Lancaster County, Pennsylvania. A 1992 novel by ► Robert James Waller centers on a photographer who is assigned to document these features in Madison County, Iowa. For 10 points, identify this kind of river crossing that is protected from rain and snow.",
    "answer": "covered bridges (prompt on \"bridges\")",
    "number": 8,
    "tournament_id": null,
    "category_id": 20,
    "subcategory_id": null,
    "round": "[missing]",
    "created_at": "2017-08-27T22:44:27.524Z",
    "updated_at": "2017-09-01T17:24:27.978Z",
    "quinterest_id": 64468,
    "formatted_text": "<b>The \"Spreuer\" example of this feature in Switzerland contains Kaspar Meglinger's 17th-century paintings of the Danse Macabre. Examples of these features from Qing Dynasty China include the \"stepping toad\" and the \"eternal celebration.\" The longest feature of this type in the world, at Hartland in New Brunswick, was damaged by ice in the 1920s. The \"Burr (*)</b>) arch\" is a typical design for these structures, which exist at Kaufman's Distillery, White Rock Forge, and Landis Mill in Lancaster County, Pennsylvania. A 1992 novel by &#9658; Robert James Waller centers on a photographer who is assigned to document these features in Madison County, Iowa. For 10 points, identify this kind of river crossing that is protected from rain and snow.",
    "formatted_answer": "covered bridges (prompt on \"bridges\")",
    "wikipedia_url": null,
    "url": "https://www.quizdb.org/api/tossups/123907",
    "type": "tossup",
    "tournament": {
    },
    "category": {
      "id": 20,
      "name": "Geography",
      "created_at": "2017-08-27T19:30:45.889Z",
      "updated_at": "2017-08-27T19:30:45.889Z",
      "url": "https://www.quizdb.org/api/categories/20"
    },
    "subcategory": {
    }
  }
];

export const mockTossup = {
"id": 123907,
  "text": "The \"Spreuer\" example of this feature in Switzerland contains Kaspar Meglinger's 17th-century paintings of the Danse Macabre. Examples of these features from Qing Dynasty China include the \"stepping toad\" and the \"eternal celebration.\" The longest feature of this type in the world, at Hartland in New Brunswick, was damaged by ice in the 1920s. The \"Burr (*) arch\" is a typical design for these structures, which exist at Kaufman's Distillery, White Rock Forge, and Landis Mill in Lancaster County, Pennsylvania. A 1992 novel by ► Robert James Waller centers on a photographer who is assigned to document these features in Madison County, Iowa. For 10 points, identify this kind of river crossing that is protected from rain and snow.",
  "answer": "covered bridges (prompt on \"bridges\")",
  "number": 8,
  "tournament_id": null,
  "category_id": 20,
  "subcategory_id": null,
  "round": "[missing]",
  "created_at": "2017-08-27T22:44:27.524Z",
  "updated_at": "2017-09-01T17:24:27.978Z",
  "quinterest_id": 64468,
  "formatted_text": "<b>The \"Spreuer\" example of this feature in Switzerland contains Kaspar Meglinger's 17th-century paintings of the Danse Macabre. Examples of these features from Qing Dynasty China include the \"stepping toad\" and the \"eternal celebration.\" The longest feature of this type in the world, at Hartland in New Brunswick, was damaged by ice in the 1920s. The \"Burr (*)</b>) arch\" is a typical design for these structures, which exist at Kaufman's Distillery, White Rock Forge, and Landis Mill in Lancaster County, Pennsylvania. A 1992 novel by &#9658; Robert James Waller centers on a photographer who is assigned to document these features in Madison County, Iowa. For 10 points, identify this kind of river crossing that is protected from rain and snow.",
  "formatted_answer": "covered bridges (prompt on \"bridges\")",
  "wikipedia_url": null,
  "url": "https://www.quizdb.org/api/tossups/123907",
  "type": "tossup",
  "tournament": {
  },
  "category": {
    "id": 20,
    "name": "Geography",
    "created_at": "2017-08-27T19:30:45.889Z",
    "updated_at": "2017-08-27T19:30:45.889Z",
    "url": "https://www.quizdb.org/api/categories/20"
  },
  "subcategory": {
  }
}