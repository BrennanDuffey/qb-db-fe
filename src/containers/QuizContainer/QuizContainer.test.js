import React, { Component } from 'react';
import { shallow } from 'enzyme';
import { QuizContainer, mapStateToProps } from '../QuizContainer';
import { mockTossups } from '../../utils/mockData';


describe('QuizContainer', () => {
  describe('QuizContainer Component', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(<QuizContainer 
        tossups={mockTossups}
      />);
    });

    it('should render to match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should render to match snapshot', () => {

    });

    it('should render to match snapshot', () => {

    });

    it('should render to match snapshot', () => {

    });
  });

  describe('mapStateToProps', () => {
    
  });
});