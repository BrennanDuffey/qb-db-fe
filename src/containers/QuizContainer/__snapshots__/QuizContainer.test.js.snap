// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuizContainer QuizContainer Component should render correctly when displayAnswer is toggled 1`] = `ShallowWrapper {}`;

exports[`QuizContainer QuizContainer Component should render to match snapshot 1`] = `ShallowWrapper {}`;
